### PR TITLE
[BEAM-8414] Cleanup Python codebase to enable some of the excluded Python lint checks.

### PR DESCRIPTION
**Pull request by stephenoken**
* Renabled the following pylint checks:
  * consider-using-set-comprehension
  * chained-comparison
  * consider-using-sys-exit
* Cleaned up the codebase to satisfy the enabled checks

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/)

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by stephenoken** [comment]
R: @pabloem, @robertwb

**Comment by stephenoken** [comment]
Run Python 2 Postcommit

**Comment by stephenoken** [comment]
Run Python PreCommit

File: sdks/python/apache_beam/io/restriction_trackers.py [hunk 1]
```diff
@@ -137,7 +137,7 @@ def try_claim(self, position):
           'of the range. Tried to claim position %r for the range [%r, %r)' %
           (position, self._range.start, self._range.stop))
 
-    if position >= self._range.start and position < self._range.stop:
+    if self._range.start >= position < self._range.stop:
```
Comment (pabloem) [inline sdks/python/apache_beam/io/restriction_trackers.py:140]:
```suggestion
    if self._range.start <= position < self._range.stop:
```
Right?

**Comment by pabloem** [comment]
retest this please

File: sdks/python/apache_beam/io/restriction_trackers.py (hunk 1 above)
Comment (stephenoken) [reply sdks/python/apache_beam/io/restriction_trackers.py:140]: oh my bad. thanks for catching that!

**Comment by stephenoken** [comment]
retest this please

**Comment by pabloem** [comment]
retest this please

**Comment by pabloem** [comment]
retest this please

**Comment by pabloem** [comment]
Unfortunately our Jenkins only listens to committers, and it's a little rickety. I'll merge once we get tests to run and pass

**Comment by pabloem** [comment]
retest this please

**Comment by stephenoken** [comment]
Ah I see. From looking at the contribution section on the beam website I thought I could trigger it. But I could totally see some bad actor abuse that. Anyways good to know!

**Comment by pabloem** [comment]
> Ah I see. From looking at the contribution section on the beam website I thought I could trigger it. But I could totally see some bad actor abuse that. Anyways good to know!

Until a couple months ago, anyone could trigger the tests, but we had to change it to be consistent with ASF policy. We're hoping to get it re-enabled for everyone soon..

**Comment by pabloem** [comment]
The linter and autoformatter are complaining. FYI:
https://builds.apache.org/job/beam_PreCommit_PythonLint_Commit/3573/console
https://builds.apache.org/job/beam_PreCommit_PythonFormatter_Commit/1391/console

**Comment by pabloem** [comment]
retest this please

**Comment by pabloem** [comment]
retest this please

**Comment by pabloem** [comment]
The formatter is complaining again: https://builds.apache.org/job/beam_PreCommit_PythonFormatter_Commit/1402/console

You can run the formatter locally with `tox -e py3-yapf` - it'll autoformat your code. Sometimes it happens to me that it errors out of execution, but still autoformats correctly, so you could try that.

File: sdks/python/apache_beam/runners/interactive/interactive_runner.py [hunk 2]
```diff
@@ -131,7 +132,7 @@ def end_session(self):
     if exit is not None:
       self._in_session = False
       _LOGGER.info('Ending session.')
-      exit(None, None, None)
+      sys.exit(None, None, None)
```
Comment (pabloem) [inline sdks/python/apache_beam/runners/interactive/interactive_runner.py:135]: Can you also revert this change? We're looking for the exit in line 131 provided by the underlying runner. I think that's causing the precommit test errors.

Comment (stephenoken) [reply sdks/python/apache_beam/runners/interactive/interactive_runner.py]: Done. I've added the `consider-using- sys-ext` to the ignored pylint list

**Comment by pabloem** [comment]
retest this please

**Comment by pabloem** [comment]
retest this please

**Comment by pabloem** [comment]
retest this please

**Comment by pabloem** [comment]
retest this please

**Comment by pabloem** [comment]
Hm lint and autoformat are complaining : (
Lint:
```
13:08:40 apache_beam/runners/interactive/interactive_runner.py:30:0: W0611: Unused import sys (unused-import)
```

you can run the autoformatter via `tox -e py3-yapf` to format your code and ensure that test will pass.

**Comment by pabloem** [comment]
retest this please

**Comment by pabloem** [comment]
alright this is ready to merge. ill wait to see if someone fixes the flink test in master

**Comment by pabloem** [comment]
Thanks for your contribution @stephenoken!

**Comment by stephenoken** [comment]
Happy to help @pabloem 😄

**Pull request merged**